### DISCUSSION TITLE
Já sei o básico, e agora?

### DISCUSSION THREAD

**Discussion by hillarysousa** [frontendbr/forum#196] [Discussão]
Oi pessoal, como estão?

Eu tenho uma duvida muito grande com relação à carreira.

Atualmente, eu sei HTML5, CSS3 e um bocado de JavaScript com jQuery, mas e agora? O que estudar?

Gostaria de opiniões e dicas de vocês :)

**Comment by leonardoelias** [comment]
O meu conselho agora e começar a fazer projetos : ) 
Colocar a mão na massa, sempre e o melhor jeito de aprender.

Se já tem alguns posta ae o/

**Comment by felipefialho** [comment]
Fala @hillaryhp, é uma questão interessante. E vai muito de encontro ao que discutimos na issue #32.

Acredito que existem vários perfis de desenvolvedor, e cada um tem suas peculiaridades e pode se especializar mais em determinados assuntos do que outros. Dificilmente um mesmo desenvolvedor consegue ser um Full-Stack que é totalmente especializado em todas as tecnologias e ciências que envolve o desenvolvimento (talvez só o @lagden :smile:).

Não sei se já está trabalhando na área. Mas como disse o @leonardoelias, faça projetos. Comece a trabalhar em Open Source, conhecer novas ferramentas e técnicas e assim ir se posicionando em alguma parte dentro desse grande universo chamado **Front-End**.

**Comment by hillarysousa** [comment]
O lance é mais sobre o que estudar, entende?

Eu me sinto meio que numa estrada com várias saídas, mas não sei por onde começar.

Eu tenho alguns freelas que fiz, mas coisa bastante básica (na minha opinião). http://hillarysousa.com

Quero fazer coisas maiores, sabe? Só não sei como.

**Comment by felipefialho** [comment]
Outra dica que pode ser interessante se está um pouco perdida é... VÁ PARA EVENTOS.

Assim você vai conhecer e trocar ideia PESSOALMENTE com diversos desenvolvedores e com certeza absoluta vai voltar para casa cheia de energia e ideias.

Escrevi esse texto, da uma lida [Por que todo desenvolvedor deve ir em eventos](https://codetalks.net/por-qual-motivo-todo-desenvolvedor-deve-ir-em-eventos-d1f276cf16b#.8ehhjcr1k).

**Comment by hillarysousa** [comment]
Eu vou até com frequência. Fui no front in Rio HTML, vou no de JS... Mas aí caimos em outra questão bem grande: timidez.

Aqui, na internet, é bem fácil pra mim discutir, opinar e conversar. Mas na vida real, só de pensar em fazer isso já me enche de vergonha.

Eu sei que isso é péssimo pra minha carreira, afinal contatos são tudo pros freelancers, mas ainda tô trabalhando pra pular essa barreira gigante (parece a muralha de gelo de GoT).

**Comment by hugodutra-zz** [comment]
Oi, @hillaryhp 

Acho que a boa se vc não sabe bem o que estudar e quer fazer coisas maiores é dominar os conceitos. Isso vai te ajudar mais a frente a aprender qualquer coisa mais rápido.

Tem esse livro do Addy Osmani de Design Patterns em Javascript que é muito bom
https://addyosmani.com/resources/essentialjsdesignpatterns/book/

**Comment by heliomsolivas** [comment]
Acho viável também pegar alguns projetos open-source para estudar como foi feito, agrega conhecimento acredito eu.

**Comment by hillarysousa** [comment]
Obrigada, gente!

**Comment by milenevlacerda** [comment]
Como uma profissional relativamente bem nova na área, vou te expor minha visão:

Comecei com o Front já faz uns 3 anos, criei muita experiência, e comecei a praticar muito e trabalhar com as tecnologias que estão em alta que sempre ouvimos falar bastante aqui no fórum ( muito importante pra ter um caminho p se guiar, o que esta em alta, o que é melhor usar e esses tipos de coisas ), mas uma coisa que me ajudou muito além de ler sobre tecnologias novas foi sim ir em eventos e ver o que o pessoal da área ( que de um modo ou de outro estão mais ativos ) andam recomendando, assim fui encontrando o que me chamava a atenção e o lado que eu estava mais gostando.

A questão sobre ser tímida: sou muito, mas acho que a necessidade de ter o conhecimento e se interessar pelas coisas abaca cobrindo tudo isso e aos poucos tu vai te soltando mais...

Fazer um mixing de tudo isso e descobrindo na área é muito importante, e mais importante ainda é descobri o que você gosta ou tem mais habilidade e se focar nisso... 😄

**Comment by vitocaio** [comment]
Bom acredito que estou na mesma situação que você, uma boa também é começar a entrar no mundo dos frameworks, sempre surge um nome mais moderno e mais simples, atualmente estou estudando Angular JS, v1x e v2.

**Comment by kelciocajueiro** [comment]
Quando eu ficar fera no JS em si, vou passar para o React e fazer umas coisinhas.

**Comment by kelciocajueiro** [comment]
Alguém tem dicas de projetos Open Source?

**Comment by felipefialho** [comment]
@kelciocajueiro Olha aqui #5

**Comment by yanmagale** [comment]
Reforçando a fala do @hugodutra , invista tempo em conceitos, patters e etc. E muito bom ter mais de uma forma de se resolver o mesmo problema, e o bacana disso é que você consegue opinar que em X cenário, tal abordagem pode ser melhor que outra.

Como bem disse o @LFeh , eventos são na minha opinião, fundamentais para qualquer desenvolvedor. Talvez seja um dos poucos ambientes que promovem conhecimento, networking, experiência, renovação e novas amizades, tudo isso num só dia.

**Comment by willianjusten** [comment]
Como o pessoal comentou e eu concordo, a melhor coisa para fixar o conhecimento e também aprender/aprimorar é criando projetinhos, sejam eles só recreativos para aprender, sejam trabalhos mais sérios.

Eu fiz um post que separo alguns conhecimentos e também uns "pequenos passos", se quiser [dar uma lida aqui](https://willianjusten.com.br/como-se-tornar-um-desenvolvedor-front-end/).

Por exemplo, você realmente já saca de JS? Tenta estudar Patterns, Testes automatizados, entender mais a fundo como o Event Loop funciona, isso vai te dar um conhecimento forte para atacar outras libs e frameworks que tem por aí.

Em HTML, você já tem conhecimento de acessibilidade (wai-aria, semântica, etc), algumas API's (web audio, video, geo, etc), seria bom dar uma olhada também.

Em CSS, você conhece padrões de arquitetura? Coisas como o ITCSS, SuitCSS, etc. Já deu uma olhada em nomenclaturas como BEM e RSCSS?

E Preprocessadores? Já deu uma estudada?

Eu diria que esses são aí os próximos passos de quem já conhece a base. O mundo front é muito grande =)

**Comment by lagden** [comment]
@LFeh o loco meu!!!!

Na verdade é muito tempo na area... since 1998...
empregos onde o chefe fala: vc sabe programar? sabe configurar o servidor? etc...

Mas vou contar o segredo para o sucesso:
Sempre torça pro time que está ganhando! :metal: 

Hoje estou no Node.js, mas já fui do Java, PHP, AS3, etc...
Andam dizendo que o Go é bom para fazer micro services...
tem outros pirando no Haskell com programação funcional...

@hillaryhp agora que você já sabe o básico, então é hora de praticar até encontrar obstáculos que dependeram de novos aprendizados. Os desafios nunca acabam, mas no futuro você resolverá eles com muita facilidade através dos skills adquiridos ao longo desse tempo...

**Comment by theandersonn** [comment]
@hillaryhp Te contar uma coisa! Eu sou tão tímido que no primeiro evento que fui, quando ia começar o intervalo do almoço eu já estava chegando na Praça XV, tamanho desespero de timidez :)
Aos poucos a gente vai se enturmando e a conversa flui naturalmente.

Apareça nos meetups de WordPress  que estão rolando no Rio.

**Comment by hillarysousa** [comment]
@theandersonn eu juro que tento ir nos meetups, mas sempre fica muito tarde pra voltar sozinha :/ daí acabo ficando com medo e não indo.

cheguei a dar ideia de fazerem streaming via periscope no Slack tem um tempinho, mas me ignoraram, daí deixei pra lá

**Comment by theandersonn** [comment]
Consideramos a ideia muito boa! E por coincidência o próprio palestrante fez a transmissão naquela ocasião. Nossa preocupação é justamente com essa questão que estamos conversando, queremos incentivar a todos que compartilhem um pouco de informação que seja, mas se sinta a vontade em palestrar. A câmera pode ser um inibidor.
Além do meetup ser um ponto de encontro pra galera se conhecer além da tela do computador.
